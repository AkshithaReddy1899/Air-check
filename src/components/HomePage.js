import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { fetchDataFromApi, searchDataFromApi } from '../redux/air';
import List from './List';

const HomePage = () => {
  const dispatch = useDispatch();
  const data = useSelector((state) => state.air);
  useEffect(() => {
    if (!data.length) {
      dispatch(fetchDataFromApi());
    }
  }, []);

  const textInput = React.createRef();
  const handleSearch = (e) => {
    e.preventDefault();
    dispatch(searchDataFromApi(textInput.current.value));
    textInput.current.value = '';
  };

  return (
    <div>
      <h1>Air checks</h1>
      <div className="search">
        <form>
          <input ref={textInput} type="text" placeholder="city" />
          <button type="submit" onClick={handleSearch}>search</button>
        </form>
      </div>
      <ul>
        {
          (data.length === 1)
            ? data.map((item) => (
              item.map((obj) => <List key={obj.id} item={obj} />)))
            : <List key={data.id} item={data} />
        }
      </ul>
    </div>
  );
};

export default HomePage;
