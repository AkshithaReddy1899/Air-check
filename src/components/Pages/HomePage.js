import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import { fetchCountryFromApi } from '../../redux/Country';

const HomePage = () => {
  const dispatch = useDispatch();
  const data = useSelector((state) => state.countries);
  console.log(data);

  const countriesToShow = [];

  data.map((item) => (
    item.map((obj) => (
      countriesToShow.push(obj)
    ))
  ));

  useEffect(() => {
    if (!data.length) {
      dispatch(fetchCountryFromApi());
    }
  }, []);

  return (
    <div>
      <h1>Air check</h1>
      <small id="error">&nbsp;</small>
      <button type="button" id="back" className="back">back</button>
      <ul id="list">
        {
          countriesToShow.map((obj) => (
            <Link to={{ pathname: `/${obj.name}`, state: 'obj.name' }} key={obj.id}>
              <button type="button">details</button>
              <h4>{obj.name}</h4>
              <p>{obj.c_code}</p>
              <p>{obj.code}</p>
            </Link>
          ))
        }
      </ul>
    </div>
  );
};

export default HomePage;
