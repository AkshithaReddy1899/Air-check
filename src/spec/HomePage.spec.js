import React from "react";
import { matches } from "./test-utils";
import { Provider } from "react-redux";
import store from "../redux/configStore";
import HomePage from '../components/Pages/HomePage';

describe("HomePage", () => {
  it('Homepage matched snapshot', () => {
   matches(<Provider store={store}><HomePage /></Provider>)
  });
});
