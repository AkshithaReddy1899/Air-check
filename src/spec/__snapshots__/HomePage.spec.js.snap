// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage Homepage matched snapshot 1`] = `
<div>
  <h1>
    Air check
  </h1>
  <small
    id="error"
  >
    Â 
  </small>
  <button
    className="back"
    id="back"
    type="button"
  >
    back
  </button>
  <ul
    id="list"
  />
</div>
`;
